src/App.tsx

```diff
@@ -1,14 +1,37 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
-import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { MemoTodoList } from './components/TodoList';
+import { MemoTodoFilter } from './components/TodoFilter';
+import { MemoTodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { filterTodos } from './services/filteredTodos';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [userTodo, setUserTodo] = useState<Todo | null>(null);
+  const [inputQuery, setInputQuery] = useState('');
+  const [selQuery, setSelQuery] = useState('all');
```

Comment:
selQuery is a bad name. It could be unclear what is `sel`. So never use short forms

But even select query is not a good choice. You can describe what exactly it is - it is status (todo status)

Also, create enum or union type for it (there are a few allowed values - it's not any string)